lib/signs/utilities/messages.ex

```diff
@@ -258,6 +267,42 @@ defmodule Signs.Utilities.Messages do
     end
   end
 
+  defp overnight_period?(current_time, config, service_ended) do
+    first_departure =
+      RealtimeSigns.headway_engine().get_first_scheduled_departure(
+        SourceConfig.sign_stop_ids(config)
+      )
+
+    last_scheduled_departure =
+      RealtimeSigns.headway_engine().get_last_scheduled_departure(
+        SourceConfig.sign_stop_ids(config)
+      )
+
+    last_actual_departure =
+      if service_ended do
```

Comment:
`service_ended` could be in the shape of a tuple if it was calculated for a sign that has multiple sources (bidirectional signs) so we'll need to keep that in mind as well

Reply:
Is this handled during the map after the `match(_, _)` and the `Tuple.to_list transformation`? Even if it is a tuple?

Reply:
Ah yes you're right, that is handled with the mapping